### PR TITLE
AB Lazy load ads

### PR DESCRIPTION
**Pull request by uplne**
Let's get wild and try lazy load of ads.

Should improve the performance as google ads script is now taking only 50ms from 150ms before + additional gain from not downloading all the ads assets on the page load.

CC @Calanthe @samdesborough

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/dfp.js [hunk 1]
```diff
@@ -190,12 +204,32 @@ define([
...
+            } else {
+                fastdom.read(function () {
+                    var scrollTop    = bonzo(document.body).scrollTop(),
+                        scrollBottom = scrollTop + bonzo.viewport().height;
 
+                    _(slots).keys().forEach(function (slot) {
+                        // if the position of the ad is above the viewport - offset (half screen size)
+                        if (scrollBottom > document.getElementById(slot).getBoundingClientRect().top + scrollTop - bonzo.viewport().height / 2) {
+                            googletag.display(slot);
+                            googletag.pubads().refresh([slots[slot].slot]);
```
Comment (samdesborough) [inline static/src/javascripts/projects/common/modules/commercial/dfp.js]: lots of slots

**Comment by gklopper** [comment]
:+1: assuming that we understand what this does to Ad impressions...

File: static/src/javascripts/projects/common/modules/commercial/dfp.js [hunk 2]
```diff
@@ -157,6 +157,14 @@ define([
             googletag.display(_.keys(slots).shift());
             displayed = true;
         },
+        displayLazyAds = function () {
+            googletag.pubads().enableSingleRequest();
```
Comment (kelvin-chappell) [inline static/src/javascripts/projects/common/modules/commercial/dfp.js]: If you `disableSingleRequest` instead, each ad will be loaded when its `display()` is called, which avoids having to make a `refresh` call.

**Comment by kelvin-chappell** [comment]
The test will need to be restricted to a particular section to make it easier to see impact on ad impressions.  
Check with Danny Doyle before deploying.

**Comment by uplne** [comment]
@kelvin-chappell So we are disabling the single request and it won't be AB test but we will run this on the 'politics' section on the US edition which should be safe area of the website to test this on.

**Comment by kelvin-chappell** [comment]
Looks good :+1:

**Pull request merged**